### PR TITLE
fix: add full client side ESM support

### PR DESCRIPTION
**Pull request by stipsan**
Clientside ESM requires import specifiers to be fully valid relative or absolute URLs. Which means we cannot leave out the file extension.

The tap tests run in the node ESM env though and don't need to include the file extension. 

I renamed the files back to .js instead of using .mjs as the tooling (like TypeScript) today supports .js far better than .mjs.

Adding `"type": "module"` to package.json was sufficient for tap to pick up on the ESM mode and work with it out of the box.

And finally, a client side test is added.

Run `npx serve .` and open http://localhost:5000/test and check the console.log.

Note that only Chrome supports importmaps natively yet: https://www.chromestatus.com/feature/5315286962012160

And you'll need to go to [chrome://flags](chrome://flags) to enable import maps specifically for the test.html stuff to work.

### CODE REVIEW

File: test.html [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<script type="importmap">
+{
+  "imports": {
+    "eventemitter3": ["https://unpkg.com/@byungi/event-emitter/dist/index.esm.js"],
```
Comment (stipsan) [inline test.html]: This alias is functionally equivalent to eventemitter3 and enables the test to run, as eventemitter3 itself does not export esm.

**Comment by trygve-lie** [comment]
Cool. 

Can we rename the test files to not contain `.test` in the file name? Tap don't care about the file names.

**Comment by digitalsadhu** [comment]
Isn't it generally better to keep .test.js so that you can include non test files in the same folder like utils or fixtures? Either that or always keep such files out of test directories.

**Comment by stipsan** [comment]
I don't have an opinion on `.test` suffixes tbh. 🙂

**Pull request merged**